src/Transport.ts

```diff
@@ -92,6 +93,39 @@ export class Transport {
     }
   }
 
+  /**
+   * Advertise a topic.
+   *
+   * @param topic The topic to advertise.
+   * @param msgTypeName The message type the topic will handle.
+   * @returns The Publisher instance.
+   */
+  public advertise(topic: string, msgTypeName: string): Publisher {
+    this.sendMessage(['adv', topic, msgTypeName, '']);
+
+    const msgDef = this.root!.lookupType(msgTypeName);
+
+    return new Publisher(
+      topic,
+      msgTypeName,
+      msgDef,
+      (topic: string, msgTypeName: string, msg: string) => {
+        this.publish(topic, msgTypeName, msg);
```

Comment:
This is a nit, but I don't think we actually need the `publish` function here.  It is just a one-liner, so we could just as easily make this:

```suggestion
        this.sendMessage(['pub_in', topic, msgTypeName, msg]);
```

Reply:
I think we could leave it as because it reduces the number of instances where the `pub_in` string is hardcoded, and its a bit more future proof if/when more functionality is added to the `publish` method.

Reply:
That's fine.  It was just a nit anyway, we can leave it as-is.